PollApp.ts

```diff
@@ -30,35 +30,97 @@ export class PollApp extends App implements IUIKitInteractionHandler {
     public async executeViewSubmitHandler(context: UIKitViewSubmitInteractionContext, read: IRead, http: IHttp, persistence: IPersistence, modify: IModify) {
         const data = context.getInteractionData();
 
-        const { state }: {
-            state: {
-                poll: {
-                    question: string,
-                    [option: string]: string,
-                },
-                config?: {
-                    mode?: string,
-                    visibility?: string,
-                },
-            },
-        } = data.view as any;
-
-        if (!state) {
-            return context.getInteractionResponder().viewErrorResponse({
-                viewId: data.view.id,
-                errors: {
-                    question: 'Error creating poll',
-                },
-            });
-        }
+        const title = data.view.title.text;
+
+        switch(title) {
+            case 'Create a poll': {
+                const { state }: {
+                    state: {
+                        poll: {
+                            question: string,
+                            [option: string]: string,
+                        },
+                        config?: {
+                            mode?: string,
+                            visibility?: string,
+                        },
+                    },
+                } = data.view as any;
+        
+                if (!state) {
+                    return context.getInteractionResponder().viewErrorResponse({
+                        viewId: data.view.id,
+                        errors: {
+                            question: 'Error creating poll',
+                        },
+                    });
+                }
+
+                if(state.config && state.config.visibility != "mixed") {
+                    try {
+                        await createPollMessage(data, read, modify, persistence, data.user.id);
+                    } catch (err) {
+                        return context.getInteractionResponder().viewErrorResponse({
+                            viewId: data.view.id,
+                            errors: err,
+                        });
+                    }
+                }
+                else {
+                    //Open mixed visibility modal
+                    try {
+                        const modal = await createMixedVisibilityModal({ question: state.poll.question, persistence, modify, data });
+                        await modify.getUiController().openModalView(modal, context.getInteractionData(), data.user);
```

Comment:
Why use the modify method to open a modal here when the context already provides an interface for sending a response modal. Checkout `context.getInteractionResponder().openModalViewResponse()`

Reply:
Using openModalViewResponse here doesn't actually open up the modal.
What's the difference between [openModalView](https://rocketchat.github.io/Rocket.Chat.Apps-engine/interfaces/accessors_iuicontroller.iuicontroller.html#openmodalview) and [openModalViewResponse](https://rocketchat.github.io/Rocket.Chat.Apps-engine/classes/uikit_uikitinteractionresponder.uikitinteractionresponder.html#openmodalviewresponse) ?

Reply:
(Future Reference) Difference between `openModalView` and `openModalViewResponse`
The first renders a new modal from scratch, however the second only updates the modal components. On using `openModalViewResponse` the first modal sticks around even after submitting the second one (which we don't need for our current use-case).

Why not use `updateModalViewResponse()`?
The modal that's to be passed to it should have the same id as the one already opened (`create-poll-modal-uuid()`) for both create poll and mixed visibility modal).
This defeats the purpose - since we're trying to act based on the modal id (impossible to differentiate if they're the same) instead of the modal title. 

Therefore, decided to go ahead with `openModalView` using `modify`.